Library/Homebrew/language/java.rb

```diff
@@ -2,7 +2,7 @@ module Language
   module Java
     def self.java_home_cmd(version = nil)
       version_flag = " --version #{version}" if version
-      "/usr/libexec/java_home#{version_flag}"
+      "#{HOMEBREW_BREW_FILE} java-home#{version_flag}"
```

Comment:
A new documented command seems pretty overkill? Could perhaps do something like `raise NotImplementedError` on the generic/non-Mac implementation? Is this used in Linuxbrew formulae and needed to work?

Reply:
`write_jar_script` uses `Language::Java.java_home_cmd`, so for that reason, it does need to work. Making it an option of an existing command doesn't seem as clean to me as a new command.

Reply:
What was the existing Linuxbrew implementation of this? We make `write_jar_script` use a more generically named function which used `java_home_cmd` only on macOS.

Reply:
The existing implementation of `java_home_cmd` returns `nil`, and the existing implementations of `java_home_env` and `overridable_java_home_env` return `{}`, which was the first version of this PR that I submitted. When `JAVA_HOME` is not defined, most tools will use the `java` interpreter found in `PATH`, which, when there is only a single Java interpreter installed, is good enough.

One place when I've seen this not work is when building a native JNI library, and the build script expects `JAVA_HOME` to be defined to find `$JAVA_HOME/include`.

Reply:
> The existing implementation of `java_home_cmd` returns `nil`, and the existing implementations of `java_home_env` and `overridable_java_home_env` return `{}`, which was the first version of this PR that I submitted.

I don't think those versions should be empty no-ops but I think a new command is a bit overkill in the opposite direction. The original implementation with `raise NotImplementedError` scattered around where appropriate feels like it may be a good call.

Reply:
This PR implements equivalent functionality to macOS, which I think is the best possible scenario. If your objection is that `brew java-home` is user-facing and documented, we can find a way to hide this tool so that it's internal to Homebrew.